src/raiden_libs/cli.py

```diff
@@ -129,3 +141,59 @@ def blockchain_options(func: Callable) -> Callable:
         func = option(func)  # type: ignore
 
     return func
+
+
+def connect_to_blockchain(
+    eth_rpc: str,
+    registry_address: Address,
+    user_deposit_contract_address: Address,
+    start_block: BlockNumber,
+    monitor_contract_address: Address,
+    contracts_version: str = None,
+):
+    try:
+        log.info(f'Starting Web3 client for node at {eth_rpc}')
+        provider = HTTPProvider(eth_rpc)
+        web3 = Web3(provider)
+        # Will throw ConnectionError on bad Ethereum client
+        chain_id = ChainID(int(web3.net.version))
+    except ConnectionError:
+        log.error(
+            'Can not connect to the Ethereum client. Please check that it is running and that '
+            'your settings are correct.',
+        )
+        sys.exit(1)
+
+    # Add POA middleware for geth POA chains, no/op for other chains
+    web3.middleware_stack.inject(geth_poa_middleware, layer=0)
+
+    # give web3 some time between retries before failing
+    provider.middlewares.replace(
+        'http_retry_request',
+        http_retry_with_backoff_middleware,
```

Comment:
Not sure if we can use this one with the MS right now, as it used `gevent.sleep` internally.

Reply:
My quick manual test indicates that `gevent.sleep` behaves just like `time.sleep` when gevent is not actively used.

Reply:
But this would mean that we block for 204.6 s with the default settings in the worst case. On the other hand we cannot do anything without connecting to the node anyways.

Reply:
Retrying for a few minutes seems like a better alternative than crashing to me.

Reply:
Makes sense 👍